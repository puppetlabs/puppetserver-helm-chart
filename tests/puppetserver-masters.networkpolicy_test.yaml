suite: test puppetserver (masters) networkpolicy creation
templates:
  - puppetserver-masters.networkpolicy.yaml
release:
  name: puppetserver
  namespace: puppet
tests:
  - it: should not create the networkpolicy by default
    asserts:
      - hasDocuments:
          count: 0
  - it: should not create the networkpolicy if the feature is not enable
    set:
      puppetserver.masters.networkPolicy.enabled: false
    asserts:
      - hasDocuments:
          count: 0

  - it: should create the networkpolicy if the feature is enable
    set:
      puppetserver.masters.networkPolicy.enabled: true
    asserts:
      - isKind:
          of: NetworkPolicy
      - equal:
          path: metadata.name
          value: puppetserver-puppetserver
