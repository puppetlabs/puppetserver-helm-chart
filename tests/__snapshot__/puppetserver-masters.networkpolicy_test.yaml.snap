manifest should match snapshot:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      labels:
        app.kubernetes.io/component: puppetserver
        app.kubernetes.io/instance: puppetserver
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: puppetserver
        app.kubernetes.io/version: 7.17.0
        helm.sh/chart: puppetserver-9.4.0
      name: puppetserver-puppetserver
    spec:
      egress:
        - ports:
            - port: 53
              protocol: TCP
            - port: 53
              protocol: UDP
          to:
            - namespaceSelector:
                matchLabels:
                  kubernetes.io/metadata.name: kube-system
              podSelector:
                matchLabels:
                  k8s-app: kube-dns
        - ports:
            - port: 53
          to:
            - ipBlock:
                cidr: 0.0.0.0/0
        - ports:
            - port: 8081
          to:
            - podSelector:
                matchLabels:
                  app.kubernetes.io/component: puppetdb
                  app.kubernetes.io/name: puppetserver
      ingress:
        - from:
            - podSelector:
                matchLabels:
                  app.kubernetes.io/name: puppetserver
          ports:
            - port: 8140
        - from:
            - namespaceSelector: {}
          ports:
            - port: 8140
      podSelector:
        matchLabels:
          app.kubernetes.io/component: puppetserver
          app.kubernetes.io/name: puppetserver
      policyTypes:
        - Egress
        - Ingress
